modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/CredentialApiUtil.java

```diff
@@ -0,0 +1,309 @@
+        } finally {
+            // Delete the temp file named credentialApi.yaml
+            if (yamlFile.exists()) {
+                log.info("Delete credentialApi temp yaml file success:{}", yamlFile
+                        .delete());
+            }
+        }
+    }
+
+    /**
+     * generate credentialApi for plugins.
+     */
+    @Async("taskExecutor")
+    public void generateCredentialApi(
+            Map<CredentialType, CredentialDefinition> credentialMap, Csp csp) {
+        createCredentialApi(credentialMap, csp);
+    }
+
+    /**
+     * delete credentialApi for plugins.
+     */
+    public void deleteCredentialApi(Csp csp) {
+        File file = new File(getCredentialApiDir(), csp + "_" + "credentialApi" + ".html");
+        if (file.exists()) {
+            log.info("Delete credentialApi html file:{}, success:{}", file.getName(),
+                    file.delete());
+        }
+    }
+
+    private Map getProMap(List<CredentialVariable> credentialVariables) {
+        Map<String, Map<String, String>> credentialVariableProMap = new HashMap<>();
+        for (CredentialVariable credentialVariable : credentialVariables) {
+            Map credentialVariableMap = new HashMap();
+            credentialVariableMap.put("example", credentialVariable.getValue());
+            credentialVariableMap.put("type", "string");
+            credentialVariableMap.put("description", credentialVariable.getDescription());
+            credentialVariableProMap.put(credentialVariable.getName(), credentialVariableMap);
+        }
+        return credentialVariableProMap;
+    }
+
+    private String getApiDocsJson(Map<CredentialType, CredentialDefinition> credentialMap,
+                                  Csp csp) {
+        String serviceUrl = getServiceUrl();
+        String variablesStr = null;
+        String variablesProStr = null;
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        CredentialDefinition credentialDefinition = credentialMap.get(CredentialType.VARIABLES);
+        List<CredentialVariable> variables = credentialDefinition.getVariables();
+        List<String> variablesNameList = variables.stream().map(CredentialVariable::getName)
+                .collect(Collectors.toList());
+        Map variableProMap = getProMap(variables);
+        String operationName = csp + "-" + credentialDefinition.getName();
+        try {
+            variablesStr = mapper.writeValueAsString(variablesNameList);
+            variablesProStr = mapper.writeValueAsString(variableProMap);
+        } catch (JsonProcessingException e) {
+            log.error("Failed to write value as string.", e);
+        }
+        //CHECKSTYLE OFF: LineLength
+        return String.format("""
+                        openapi: 3.0.1
+                        info:
+                          title: %s plugin credentials definition
+                          version: v1.16
```

Comment:
what is this version?

Reply:
now, this version is meaningless, just a specification for OpenApi.

Reply:
in this case, let us put the xpanse application version.  `app.version` should have this value. We were using this in the banner.txt also.